include/meshio/obj.hpp

```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright (c) 2017, Lakshman Anumolu, Pradeep Garigipati
+ * All rights reserved.
+ *
+ * This file is part of MeshIO whose distribution is governed by
+ * the BSD 2-Clause License contained in the accompanying LICENSE.txt
+ * file.
+ */
+
+#ifndef MESHIO_OBJ_OBJ_HPP_
+#define MESHIO_OBJ_OBJ_HPP_
+
+#define TINYOBJLOADER_IMPLEMENTATION
+
+#include "./utilities.hpp"
+#include "./meshio_defines.hpp"
+#include "../third_party/tiny_obj_loader.h"
+
+#include <iostream>
+#include <fstream>
+#include <sstream>
+
+namespace meshio
+{
+
+/* Datastructure to store Wavefront OBJ */
+template<class T>
+class OBJAttribute : public GeometryData<T> {
+  public:
+    std::string mOBJName;
+    std::vector<int> mMaterialIDs;
+
+    OBJAttribute() {}
+    ~OBJAttribute() {}
+
+    void resizeVertexAttributes(unsigned pNumTriangles) {
+        this->mPositions.resize(pNumTriangles);
+        this->mNormals.resize(pNumTriangles);
+        this->mTexturePositions.resize(pNumTriangles);
+        this->mColors.resize(pNumTriangles);
+        this->mParameterSpaceVertices.resize(pNumTriangles);
+    }
+
+    void resizeIndices(unsigned pNumIndices) {
+        this->mTriVertexIndices.resize(pNumIndices);
+        this->mTriTextureIndices.resize(pNumIndices);
+        this->mTriNormalIndices.resize(pNumIndices);
+    }
+
+    void clear() {
+        GeometryData<T>::clear();
+        mMaterialIDs.clear();
+    }
+
+    bool operator==(const OBJAttribute<T>& pOBJAttribute) const {
+        if (dynamic_cast<GeometryData<T>&>(
+                        const_cast<OBJAttribute<T>&>(*this)) ==
+                dynamic_cast<GeometryData<T>&>(
+                        const_cast<OBJAttribute<T>&>(pOBJAttribute))) {
+
+            if (not mOBJName.compare(pOBJAttribute.mOBJName)) {
+                // Checking equality for mMaterialIDs.
+                if (mMaterialIDs.size() && pOBJAttribute.mMaterialIDs.size()) {
+                    for (unsigned i = 0; i < mMaterialIDs.size(); ++i) {
+                        if (not (mMaterialIDs[i] == pOBJAttribute.mMaterialIDs[i])) {
+                            return false;
+                        }
+                    }
+                    return true;
+                } else if (not mMaterialIDs.size() &&
+                           not pOBJAttribute.mMaterialIDs.size()) {
+                    return true;
+                } else {
+                    return false;
+                }
+
+            } else {
+                return false;
+            }
+
+        } else {
+            return false;
+        }
+    }
+
+};
+
+template<class T>
+class OBJData {
```

Comment:
@9prady9 

* Here, we need two vectors (i) that stores geometric attributes, (ii) material information. If I try to use one derived class from `GeometryData<T>`, to store entire obj information, I am unable to differentiate geometry and material information that we get from tinyobjloader. So, I introduced one derived class of `GeometryData<T>` and used vector of this derived class in `OBJData`. So, instead of using something similar to `vector<STLData>`, I used `vector<mOBJAttributes>`. It is getting difficult for me to use one class in this case.